### PR TITLE
Rewrite the SensorFactory

### PR DESCRIPTION
**Pull request by TheDevMinerTV**

### CODE REVIEW

**Review by Eirenliel — CHANGES_REQUESTED** [review]
Good stuff but needs changes ^^

**Comment by TheDevMinerTV** [comment]
ℹ️ Ready for QA (again)

**Comment by TheDevMinerTV** [comment]
ℹ️ Ready for review (again)

File: src/globals.h [hunk 1]
```diff
@@ -29,7 +29,7 @@
 #include "defines.h"
 
 #ifndef SECOND_IMU
-#define SECOND_IMU IMU
+#define SECOND_IMU IMU_NONE
```
Comment (Eirenliel) [inline src/globals.h:32]: I don't think it should default to NONE.

File: src/consts.h [hunk 2]
```diff
@@ -24,6 +24,7 @@
 #define SLIMEVR_CONSTS_H_
 
 // List of constants used in other places
+#define IMU_NONE 0
```
Comment (Eirenliel) [inline src/consts.h:27]: I don't think NONE is needed.

**Review by Eirenliel — CHANGES_REQUESTED** [review]

File: src/sensors/ErroneousSensor.cpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        }
+
+        void ErroneousSensor::sendData()
+        {
+            if (m_SentFirstData)
+            {
+                return;
+            }
+
+            Network::sendRotationData(&quaternion, DATA_TYPE_NORMAL, 255, sensorId);
```
Comment (Eirenliel) [inline src/sensors/ErroneousSensor.cpp:43]: This is not what it needs to send te be displayed, it should just return `SensorStatus::SENSOR_ERROR` in `getSensorState()`, everything else will be handled.

**Review by Eirenliel — CHANGES_REQUESTED** [review]

**Review by unlogisch04 — CHANGES_REQUESTED** [review]
Correct IMU debug text, and move motionLoop to a other position

File: src/sensors/SensorManager.cpp [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+#elif IMU == IMU_ICM20948
+                    m_Sensor1 = new ICM20948Sensor(0, firstIMUAddress, IMU_ROTATION);
+#endif
+                }
+
+                m_Sensor1->motionSetup();
+            }
+
+            {
+#if SECOND_IMU == IMU_BNO080 || SECOND_IMU == IMU_BNO085 || SECOND_IMU == IMU_BNO086
```
[the remark this answers is not in the document]
Comment (TheDevMinerTV) [reply src/sensors/SensorManager.cpp:82]: Can't happen, this is solved by the redefine in `src/globals.h`

**Pull request merged**